asreview/compat.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2019 The ASReview Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def convert_id_to_idx(data_obj, record_id):
+    """Convert record_id to row number."""
+
+    inv_record_id = dict(zip(data_obj.df.index.tolist(), range(len(data_obj))))
+
+    result = []
+    for i in record_id:
+        try:
+            result.append(inv_record_id[i])
+        except KeyError:
+            raise KeyError(f"record_id {i} not found in data.")
```

Comment:
So if a record_id already exists in the data you will get a warning? or will the exisiting record_id be renamed to record_id_old?

Reply:
If there is a record_id available in the dataset, it will be used as record_id. There is no need to use a deprecated or old version of the record_id, because it is a persistent identifier from now on.